src/react/App.jsx

```diff
@@ -22,13 +22,63 @@ import AlertsProvider from './AlertManager';
 import Router from './Router';
 import Snackbar from './ui/Snackbar';
 
+function parseDashboardConfig(json) {
+  const {
+    astarte_api_url,
+    realm_management_url,
+    appengine_url,
+    pairing_url,
+    flow_url,
+    enable_flow_preview,
+    default_realm,
+    default_auth,
+    auth,
+  } = json;
+
+  if (!astarte_api_url || !auth || !default_auth) {
+    return null;
+  }
+
+  const config = {
+    astarte_api_url,
+    realm_management_url,
+    appengine_url,
+    pairing_url,
+    flow_url,
+    enable_flow_preview,
+    default_realm,
+    default_auth,
+    auth: {},
+  };
+
+  auth.forEach((authOption) => {
+    if (authOption.type === 'token') {
+      config.auth.token = {
+        enabled: true,
+      };
+    } else if (authOption.type === 'oauth') {
+      config.auth.oauth = {
+        enabled: true,
+        oauth_api_url: authOption.oauth_api_url,
+      };
+    }
+  });
+
+  const defaultAuth = config.auth[default_auth];
+  if (!defaultAuth || !defaultAuth.enabled) {
+    return null;
+  }
+
+  return config;
```

Comment:
Do you think we could somehow have a single `auth` object that includes the indication of the default auth method together with the available methods?

Reply:
Not really. In fact those are 2 different things, one is the definition of what's available and what's not, on the other hand you have a "preference".
Merging them in a single object could lead to some edge case like "what if 2 methods are flagged as default at the same time?"